source/common/common/utility.h

```diff
@@ -132,16 +134,74 @@ class StringUtil {
   static uint32_t itoa(char* out, size_t out_len, uint64_t i);
 
   /**
-   * Trim trailing whitespace from a string in place.
+   * Trim leading whitespace from a string view.
+   * @param source supplies the string view to be trimmed.
+   * @return trimmed string view.
    */
-  static void rtrim(std::string& source);
+  static absl::string_view ltrim(absl::string_view source);
 
   /**
-   * Size-bounded string copying and concatenation
+   * Trim trailing whitespaces from a string view.
+   * @param source supplies the string view to be trimmed.
+   * @return trimmed string view.
    */
-  static size_t strlcpy(char* dst, const char* src, size_t size);
+  static absl::string_view rtrim(absl::string_view source);
+
+  /**
+   * Trim leading and trailing whitespaces from a string view.
+   * @param source supplies the string view to be trimmed.
+   * @return trimmed string view.
+   */
+  static absl::string_view trim(absl::string_view source);
 
   /**
+   * Look up for an exactly token in a delimiter-separated string view.
+   * @param source supplies the delimiter-separated string view.
+   * @param multi-delimiter supplies chars used to split the delimiter-separated string view.
+   * @param token supplies the lookup string view.
+   * @param trim_whitespace remove leading and trailing whitespaces from each of the split
+   * string views; default = true.
+   * @return true if found and false otherwise.
+   *
+   * E.g,
+   *
+   * find("A=5; b", "=;", "5")   . true
+   * find("A=5; b", "=;", "A=5") . false
+   * find("A=5; b", "=;", "A")   . true
+   * find("A=5; b", "=;", "b")   . true
+   * find("", ".", "")           . true (spliting "" with "." returns "" which contains "")
```

Comment:
I still think this is a weird case, and wonder what you have in mind for defining the semantics this way.

Reply:
👀

Reply:
If the source cannot be split by any of the provided delimiters, `absl::StrSplit(source, absl::ByAnyChar(delimiters))` returns a vector with one element, which contains the source itself. So, in the above doc `find("", ".", "")` returns true since ("" equals ""). I guess I should just make the example a bit more clear. Does that make sense?

Reply:
OK, sounds good.  Let's stick with what you have.